File: source/includes/connect/mongoclient.c
#include <mongoc/mongoc.h>
#include <bson/bson.h>

int main (void) {

  // start-connect-to-atlas-w-uri
  // Initialize the C Driver
  mongoc_init ();

  // Create a new client and connect to the server
  mongoc_client_t *client = mongoc_client_new ("mongodb+srv://<db_username>:<db_password>@<hostname:port>/?<options>");
  mongoc_database_t *database = mongoc_client_get_database (client, "admin");

  bson_t *ping = BCON_NEW ("ping", BCON_INT32 (1));
  bson_t reply = BSON_INITIALIZER;
  bson_error_t error;

  // Send a ping to confirm a successful connection
  if (!mongoc_client_command_simple (client, "admin", ping, NULL, &reply, &error)) {
    fprintf(stderr, "error: %s\n", error.message);
    goto cleanup;
  }

  printf ("Pinged your deployment. You successfully connected to MongoDB!\n");

  cleanup:
    bson_destroy (&reply);
    bson_destroy (ping);

    mongoc_database_destroy (database);
    mongoc_client_destroy (client);

    // Cleanup the C Driver
    mongoc_cleanup ();
  // end-connect-to-atlas-w-uri
}

File: content/kotlin-sync/upcoming/source/includes/connect/mongoclient.kt
import com.mongodb.*
import com.mongodb.kotlin.client.MongoClient
import org.bson.BsonInt64
import org.bson.Document

fun main() {
    
    // start-connect-to-atlas
    // Replace the placeholder with your Atlas connection string
    val uri = "<connection string>"
    val mongoClient1 = MongoClient.create(uri)
    // end-connect-to-atlas-w-uri
    
    // Construct a ServerApi instance using the ServerApi.builder() method
    val serverApi = ServerApi.builder()
        .version(ServerApiVersion.V1)
        .build()
    val settings = MongoClientSettings.builder()
        .applyConnectionString(ConnectionString(uri))
        .serverApi(serverApi)
        .build()

    // Create a new client and connect to the server
    val mongoClient = MongoClient.create(settings)
    val database = mongoClient.getDatabase("sample_mflix")

    try {
        // Send a ping to confirm a successful connection
        val command = Document("ping", BsonInt64(1))
        val commandResult = database.runCommand(command)
        println("Pinged your deployment. You successfully connected to MongoDB!")
    } catch (me: MongoException) {
        System.err.println(me)
    }
    // end-connect-to-atlas
}
